### DISCUSSION TITLE
Uprade smithy4s 0.17.1 -> 0.17.2. Got time parsing error

### DISCUSSION THREAD

**Discussion by Quafadas** [disneystreaming/smithy4s#718] [Q&A] [answered]
Moving from smithy4s 0.17.1 to 0.17.2 might get you this; 

```
Unknown time-zone ID: Europe/Paris
java.time.zone.ZoneRulesProvider$.getProvider(http://localhost:5173/@fs/C:/temp/ilsclaims/out/frontend/fastLinkJS.dest/internal-ba36307f7b0aa85310c10f637442996751b8bf32.js:4568:13)
```

See here; 

https://github.com/plokhotnyuk/jsoniter-scala/pull/977

Solution : add this to your build; 
```scala
ivy"io.github.cquiroz::scala-java-time-tzdb::2.5.0"
```
Recorded here for posterity... 

Thanks to velvetbaldmine :-).

**Comment by Baccata** [comment, marked as answer]
@Quafadas just to be clear : you're getting this error because the dependency was pulled transitively and it was used somewhere in your application, correct ?

**Reply by Quafadas** [reply]
You know that much better than I! And yes... my understanding is that the pull request to jsoniter removes the java-scala-tzdb, which (transitively) causes that problem by no longer being available to java-time's stringly typed api... as you point out. 

I'd certainly accept the argument, that reporting this as a "smithy" issue per se, is misleading... as it kind of... isn't.

On the other hand: me caveman, change smithy version, get error. 

In userland, it's is a really scary error too - you get an obfuscated JS stack trace at runtime deep in _library_ code so logging (my current js "debugging" solution) is useless. To identify the solution without knowing the solution, you would need to bypass three pretty nasty obstacles; 

1. Identifying the offending API call in the absence of source code (source map support in scala-js land right is spotty at best, I suspect a lot of libraries havne't changed the compile flag from scala 2-> 3, so this is a fairly systemic problem). As you know I'm filing bugs and PRs, and finding out how bad I suck at sbt, but the scala 3 ecosystem supporting JS sourcemaps is probably beyond my influence and will take time
2. A stringly typed API - so no compiler quarterbacking for you
3. Transitive dependancy change.

If I'd been doing this in a months time, or migrating more than one minor version of one library, this might have gotten (very) painful - I read the error and knew I was stuffed... and consider very lucky that Anton popped up and knew the answer! My motivation in writing this out the above, is attempting to be a good citizen by leaving the breadcrumbs behind :-).

**Reply by Baccata** [reply]
Yup, that's totally fine, I'm just ensuring that it's nothing in how this library uses jsoniter that was the problem. You are absolutely right to create a discussion around it.

**Reply by plokhotnyuk** [reply]
AFAIK, `smithy4s-json` and `Timestamp` from `smithy4s-core` don't use time-zones.

@Quafadas I'll happy to fix unwanted dependency on time-zones in `jsoniter-scala-core` if you really don't use them directly.

**Reply by Quafadas** [reply]
@plokhotnyuk I'm personally happy with the status quo. Also, I'm not the person who should be driving such decisions as I lack the knowledge for them to be properly informed - my contribution is simply a thanks for your excellent library.

**Reply by kubukoz** [reply]
> I'd certainly accept the argument, that reporting this as a "smithy" issue per se, is misleading... as it kind of... isn't.

You made a discussion entry in the Q&A section, to me that's a perfectly valid thing to do, quite appreciated actually, as people may be hitting this after you :)

**Reply by daddykotex** [reply]
@Quafadas is the best kind of user
